### ISSUE TITLE
[feature] Subcommands

### ISSUE THREAD

**Issue by TimWhiting** [frencojobs/derry#2]
I think it would be really nice to have a way to specify dependencies on other commands

For example:
I have this, but a little bit more complex. 
```yaml
scripts:
  prep:
   - step 1
   - step 2
   - step 3
  run:
   - derry run prep
   - flutter run
  build:
   - derry run prep
   - flutter build
```

Anyways, you can see that multiple commands run derry as part of the process in order to run previously needed steps.
A way to specify them so that it all happens as part of one flow, rather than another derry process being started as a subprocess would be nice. Essentially something like this:
```yaml
  prep:
   - step 1
   - step 2
   - step 3
  run:
   - @prep
   - flutter run
  build:
   - @prep
   - flutter build
```
That would desugar to just this as part of a preprocessing step:

```yaml
  prep:
   - step 1
   - step 2
   - step 3
  run:
   - step 1
   - step 2
   - step 3
   - flutter run
  build:
   - step 1
   - step 2
   - step 3
   - flutter build
```

It could be as simple as checking for the @ sign and then seeing if any commands match that name, and then replacing the command with the @ sign with the list of commands from the rule that matches that name.

**Comment by frencojobs** [comment]
Wow! That is a really good idea. I would love to do it.

**Comment by frencojobs** [comment]
@TimWhiting are **@** symbols valid in YAML. I don't know much about YAML, but using these **@** symbols are causing a Parsing Error.

**Comment by TimWhiting** [comment]
I'm not sure. That was just an idea. Probably a different symbol would be better

**Comment by frencojobs** [comment]
According to this,
> Use quotes in YAML if your value includes special characters. For example, these special characters may require quotes: {, }, [, ], ,, &, :, *, #, ?, |. -, <. >, =, !, %, @, \.

using one of these characters would be required to use **quotes**, and I don't think that looks fancy as it could be.

So I'm thinking to do something like `sub:prep` instead of `"@prep"`. In your example, it would be like
```yaml
prep:
   - step 1
   - step 2
   - step 3
  run:
   - sub:prep
   - flutter run
  build:
   - sub:prep
   - flutter build
```
How do you think?

**Comment by TimWhiting** [comment]
I like the colon idea.
I'm not so sure about the prefix sub: I think the prefix `ref:` is a bit nicer.
Also it looks like : still is a special character according to what you just quoted. So it might force it to be:
```yaml
 - "ref:prep"
```
I agree that we should not have to quote it.
How about something like:
```yaml
 - $prep
```
or this:
```yaml
 - ref$prep
```
With the prefix you have more options to adding other features using that same convention, you just have to notify users that a $ in the middle of a command is reserved unless in quotes

**Comment by frencojobs** [comment]
I think the dollar sign idea is way more appealing than others. We can add `\` to un-parse dollar sign for commands that are not subcommands.
```yaml
- $prep # for subcommand prep
- \$prep # if the $prep is a command

**Comment by frencojobs** [comment]
Subcommands should be available in v0.0.2 ✌️. Thank you! @TimWhiting for idea & contribution. Feel free to reopen the issue if the implementation was incorrect or wasn't as intended.
